GetStream/stream-chat-swift
# Mark As Unread: Add UI and WS Events Handling

### 🔗 Issue Links

#2519 

### 🎯 Goal

Show unread messages separator when marking a message as unread

### 📝 Summary

This PR adds the UI related part for Mark as unread. It also adds handling for the WS Events that are sent when this happens

### 🛠 Implementation

- Reuses `ChatThreadRepliesCountDecorationView` to show the UI separator.
    - Class has been renamed to `ChatUnreadMessagesCountDecorationView` so it is generic for Threads and Unread messages
- Mark as unread WS Events are now handled through `ChannelReadUpdaterMiddleware`
- Disable Mark as unread from Threads

### 🎨 Showcase
<img src="https://user-images.githubusercontent.com/7887319/223991028-92eac755-d36c-4471-98b1-00446a7be94d.png" width="200">

### 🧪 Manual Testing Notes

1. Go to a channel
2. Long press on a message
3. Choose "Mark as unread"

Expected result:

- Should add a UI indicator on top of the message
- The message should be marked as unread

### ☑️ Contributor Checklist


### 🎁 Meme

![](https://media.giphy.com/media/3ohjUOCrQdfCh9qVDa/giphy.gif)
